qt-style.css

```diff
@@ -1,140 +1,107 @@
-MainWindow #translateLine, ScanPopup #translateLine, MainWindow #wordList, MainWindow #dictsPane #dictsList, MainWindow #historyPane #historyList, MainWindow #favoritesPane #favoritesTree
-{
+MainWindow #translateLine, ScanPopup #translateLine, MainWindow #wordList, MainWindow #dictsPane #dictsList, MainWindow #historyPane #historyList, MainWindow #favoritesPane #favoritesTree {
 }
 
-MainWindow #translateLine[noResults="true"],
-ScanPopup #translateLine[noResults="true"] {
+#translateLine {
+    color: palette(Text); /* ensure the text is inverse of the background, and also placehold text is grey */
+}
+
+#translateLine[noResults="true"] {
     background: #febb7d;
 }
 
 #navToolbar #menuButton::menu-indicator {
-  image: none;
+    image: none;
 }
 
-#navToolbar #translateLine
-{
-  border: 1px solid gray;
-  border-radius: 3px;
-  padding: 1px 3px 2px 3px;
-  border-top-left-radius: 0px;
-  border-bottom-left-radius: 0px;
-  margin-left: 0px;
+#navToolbar #translateLine {
+    border: 1px solid gray;
+    border-radius: 3px;
+    padding: 1px 3px 2px 3px;
+    border-top-left-radius: 0;
+    border-bottom-left-radius: 0;
+    margin-left: 0;
 }
 
-#navToolbar GroupComboBox
-{
-  margin-right: 0px;
-  border: 1px solid gray;
-  border-right: 1px transparent;
-  border-radius: 3px;
-  border-top-right-radius: 0px;
-  border-bottom-right-radius: 0px;
-  padding: 2px 4px 2px 4px;
-}
+GroupComboBox {
+    margin-right: 0;
+    border: 1px solid gray;
+    border-right: 1px transparent;
+    border-radius: 3px;
+    border-top-right-radius: 0;
+    border-bottom-right-radius: 0;
+    padding: 2px 4px 2px 4px;
 
-#navToolbar GroupComboBox:on {
-  padding: 3px 3px 1px 5px;
+    background: palette(window); /* Distinguish the combobox from the translate line by make it darker*/
+    color: palette(Text);
 }
 
-#navToolbar GroupComboBox::drop-down {
-   subcontrol-origin: padding;
-   subcontrol-position: top right;
-   width: 15px;
-   border-left: 1px solid darkgray;
-   border-right: 1px transparent;
-}
-
-#navToolbar GroupComboBox::down-arrow {
-    image: url(:/icons/1downarrow.svg);
-    max-height: 100%;
+GroupComboBox:on {
+    padding: 3px 3px 1px 5px;
 }
 
-QTabBar::close-button {
-  image: url(:/icons/closetab.svg)
-}
-QTabBar::close-button:hover {
-  image: url(:/icons/closetab-hover.png)
-}
+GroupComboBox::drop-down {
+    subcontrol-origin: padding;
+    subcontrol-position: top right;
+    width: 15px;
+    border-left: 1px solid darkgray;
+    border-right: 1px transparent;
 
+    /* Hack: For unknown reason, if the padding right is unset, the drop-down won't show full content */
```

Comment:
To deal with this bug, the dropdown cannot be shown after using dark mode.

https://forum.freemdict.com/t/topic/11495/1883

I think there are some mixing problems when using CSS, but that probably requires us to get rid of the CSS.